test_pull_requests

```diff
@@ -1189,8 +1362,44 @@ popd
         # comment to reflect the new state
         if submit_test_job
           delete_comment_with_prefix(id, base_repo, ACTION_PREFIX, comments)
-          extended_tests = get_extended_tests(req, comments, branch, settings)
-          submit_tests(repo_to_pull_request, base_repo, branch, comment_id, extended_tests, settings)
+          # Check for pretest_settings_key, so we might skip a round
+          # of tests prior to merge
+          pretest_settings = Properties['settings'][settings['pretest_settings_key']]
+          repo_to_comment_fields = {}
+          can_skip_merge_tests = false
+          if pretest_settings
+            pretest_prefix = pretest_settings['test_prefix']
+            can_skip_merge_tests = repo_to_pull_request.all? do |repo, pull_request|
+              pretest_comment = get_comment_with_prefix(pull_request['number'], repo, pretest_prefix, (repo == base_repo ? comments : nil))
+              if pretest_comment.nil?
+                false
+              else
+                pretest_fields = extract_bot_comment_fields(pretest_comment['body'], pretest_settings)
+                p_base_commit = base_commit_tag_to_sha(pretest_fields[:base_commit])
```

Comment:
If we're just copying it into a new comment, we don't need to extract and then recompose the whole field